README.md

```diff
@@ -19,14 +19,188 @@ The cleaned must-gather can then be found in the `must-gather-output-cleaned` fo
+
+There is another type of replacement called `Consistent`, that can be configured like this:
+
+```
+config:
+  obfuscate:
+  - type: MAC
+    replacementType: Consistent
+```
+
+This will detect all MAC addresses and replace them with a "consistent" identifier that looks like this `xxx-mac-000001-xxx`. 
+Let's say one of your eth interfaces has the mac address `52:54:00:5e:ee:c6` and was logged, then `must-gather-clean`  will guarantee that it will always be assigned the same obfuscated consistent identifier across all files in a must-gather. 
+That primarily helps our support and engineers to ensure we can still understand and reproduces challenges that you were facing without putting your classified information at risk.
+
+Let's look at another obfuscation type named `IP`. This type can be used to clean IP addresses, we support both IPv4 and IPv6 except for local interfaces (`127.0.0.1`, `0.0.0.0` and `::1`) that will be preserved.
+
+You can configure this along with the MAC obfuscator like this:
+
+```
+config:
+  obfuscate:
+  - type: MAC
+    replacementType: Consistent
+  - type: IP
+    replacementType: Consistent
+```
+
+On a line-by-line basis, this will always execute the MAC obfuscation first and then the IP obfuscator - we'll go through this behaviour in more detail in the `Chaining obfuscators and side effects` section below. 
+
+Another configuration flag that we support for each obfuscation type is the `target`. The target is useful when the confidential information can be found not only in the file content, but also in the folder or file names. 
+This can very frequently happen with IP addresses, for example through node names. You can control that independently for each type like this:
+
+```
+config:
+  obfuscate:
+  - type: MAC
+    replacementType: Consistent
+    target: FileContents
+  - type: IP
+    replacementType: Consistent
+    target: FilePath
+```
+
+As you can see, the MAC obfuscator would work on file content whereas the IP obfuscator would work only on FilePaths. There is a mixed target called `All`, that will obfuscate on both paths and contents. 
+The default if no target is specified is `FileContents`, it is thus always recommended using the IP obfuscator with `target: All` to not accidentally leak IP information through folder names.
+
+The third built-in type of obfuscation is `Domain`, let's take a look how this can be configured:
+
+```
+config:
+  obfuscate:
+  - type: Domain
+    domainNames:
+    - "rhcloud.com"
+    - "dev.rhcloud.com"
+```
+
+As you can see, this type must be customized by supplying domain names. 
+Kubernetes resources are defined along with their domain names (eg "apps.openshift.io/v1") and thus would be automatically recognized as such and obfuscated as a false-positive.
```

Comment:
replace eg with e.g. across document (though as per doc guidelines in product docs we use 'for example' and refrain e.g.)

Reply:
removed eg everywhere